src/TagsInput/TagsInput.js

```diff
@@ -0,0 +1,151 @@
+  static defaultProps = {
+    fontSize: 16,
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      value: props.value ?? '',
+      containerWidth: null,
+      inputWidth: null,
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
+    const { value } = this.props;
+    if (value !== prevProps.value) {
+      this.setState({ value });
+    }
+  };
+
+  setContainerWidth = (event: ViewLayoutEvent) => {
+    this.setState({ containerWidth: event.nativeEvent.layout.width });
+  };
+
+  setInputWidth = (event: ViewLayoutEvent) => {
+    this.setState({ inputWidth: event.nativeEvent.layout.width });
+  };
+
+  setRef = (input: React.Element<any>) => {
+    this.inputRef = input;
+  };
+
+  getWidth = (width: ?number) => {
+    const { containerWidth } = this.state;
+    if (containerWidth && width) {
+      return containerWidth - width;
+    }
+    return undefined;
+  };
+
+  handleChange = (value: string) => {
+    const { value: oldValue } = this.state;
+    const { onChangeText } = this.props;
+    if (value !== oldValue) {
+      this.setState({ value });
+      onChangeText(value);
+    }
+  };
+
+  handleClear = () => {
+    const { onClearPress } = this.props;
+    this.setState({ value: '' });
+    Platform.OS === 'web' && this.inputRef.focus();
+
+    if (onClearPress) {
+      onClearPress();
+    }
+  };
+
+  render() {
+    const { selected, fontSize, label, placeholder, disabled } = this.props;
+    const { value, inputWidth } = this.state;
+
+    return (
+      <View style={styles.container}>
+        <Text weight="bold" style={(styles.label, { fontSize })}>
+          {label}
+        </Text>
+        <View style={styles.fieldContainer} onLayout={this.setContainerWidth}>
+          <TagsContainer
+            maxWidth={this.getWidth(inputWidth)}
+            tags={selected}
+            fontSize={fontSize}
+          />
+          <InputField
+            setRef={this.setRef}
+            fontSize={fontSize}
+            disabled={disabled}
+            minWidth={MIN_WIDTH}
+            maxWidth={this.getWidth(MIN_WIDTH)}
+            onLayout={this.setInputWidth}
+            value={value}
+            placeholder={
+              selected?.length === 0 && placeholder ? placeholder : null
+            }
+            onChangeText={this.handleChange}
+          />
+        </View>
+        <DeleteButton
+          style={{ opacity: value.length > 0 ? 1 : 0 }}
+          onPress={this.handleClear}
+          disabled={!value.length ?? disabled}
+        />
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  fieldContainer: {
+    flexDirection: 'row',
+    flex: 1,
+    marginStart: 4,
+    overflow: 'hidden',
+  },
+
+  label: {
+    color: defaultTokens.paletteInkDark,
+  },
+});
+
+export default TagInput;
```

Comment:
Just to be consistent with other components, can you put the export default where you define the class?